cmp/form/JsonField.js

```diff
@@ -57,41 +88,25 @@ export class JsonField extends HoistField {
     }
 
     createJsonEditor(taCmp) {
-        const editorSpec = {
-            theme: this.darkTheme ? 'dracula' : 'default',
-            mode: 'application/json',
-            lineWrapping: this.props.lineWrapping || false,
-            lineNumbers: true,
-            autoCloseBrackets: true,
-            extraKeys: {
-                'Cmd-P': this.onFormatKey,
-                'Ctrl-P': this.onFormatKey
-            },
-            foldGutter: true,
-            scrollbarStyle: 'simple',
-            gutters: [
-                'CodeMirror-linenumbers',
-                'CodeMirror-foldgutter'
-            ],
-            readOnly: this.props.disabled,
-            lint: true
-        };
-        
-        const props = this.props,
-            taDom = ReactDOM.findDOMNode(taCmp),
+        const {editorProps, disabled, width, height} = this.props,
+            editorSpec= defaultsDeep(
+                editorProps,
+                JsonField.defaultEditorProps,
+                {theme: this.darkTheme ? 'dracula' : 'default', readOnly: disabled}
+            );
+
+        const taDom = ReactDOM.findDOMNode(taCmp),
             editor = codemirror.fromTextArea(taDom, editorSpec);
 
         editor.on('change', this.handleEditorChange);
         editor.on('focus',  this.onFocus);
         editor.on('blur',  this.onBlur);
         editor.on('keyup',  this.onKeyUp);
 
-        let {height, width} = props;
-        if (!(isUndefined(height) && isUndefined(width))) {
```

Comment:
defaultProps also allows us to not do this business. Unless there is a reason to jump through these hoops, the new code is much easier to read. (But perhaps we WANT undefined sometimes, or perhaps we want to set size when one is null and the other is undefined? Or both are null? Perhaps Colin could shed some light here?).